server/middleware/roleMiddleware.test.ts

```diff
@@ -4,26 +4,57 @@ import { userFactory } from '../testutils/factories'
 import { UnauthorizedError } from '../utils/errors'
 import { createRoleMiddleware } from './roleMiddleware'
 
+import config from '../config'
+
 describe('createRoleMiddleware', () => {
-  it('returns middleware that calls the handler when the user has the required role', async () => {
+  const originalValue = config.flags.allRegionsEnabled
+
+  afterEach(() => {
+    config.flags.allRegionsEnabled = originalValue
+  })
+
+  it('returns middleware that calls the handler when the user has the required role and region', async () => {
     const handler = jest.fn()
     const request = createMock<Request>()
     const response = createMock<Response>({
       locals: {
         user: userFactory.build({
-          roles: ['referrer'],
+          roles: ['cas3_referrer'],
+          region: { name: 'Kent, Surrey & Sussex' },
         }),
       },
     })
     const next = jest.fn()
+    config.flags.allRegionsEnabled = false
 
-    const middleware = createRoleMiddleware('referrer')
```

Comment:
I feel like the change to `cas3_referrer` is worth its own commit

Reply:
remove cas3_referrer based on the clarification that the backend api sends referrer and not cas3_referrer